packages/mux-player/src/index.ts

```diff
@@ -0,0 +1,398 @@
+  }
+
+  get playerSoftwareName() {
+    return playerSoftwareName;
+  }
+
+  get playerSoftwareVersion() {
+    return playerSoftwareVersion;
+  }
+
+  get hls() {
+    return this.video?.hls;
+  }
+
+  get mux() {
+    return this.video?.mux;
+  }
+
+  /**
+   * Get Mux asset playback id.
+   * @return {string?}
+   */
+  get playbackId() {
+    return getVideoAttribute(this, MuxVideoAttributes.PLAYBACK_ID);
+  }
+
+  /**
+   * Mux Data env key
+   * @return {string?}
+   */
+  get envKey() {
+    return getVideoAttribute(this, MuxVideoAttributes.ENV_KEY);
+  }
+
+  /**
+   * Get video engine debug flag.
+   * @return {boolean}
+   */
+  get debug() {
+    return getVideoAttribute(this, MuxVideoAttributes.DEBUG) != null;
+  }
+
+  /**
+   * Set video engine debug flag.
+   * @param  {boolean} val
+   */
+  set debug(val) {
+    if (val) {
+      this.setAttribute(MuxVideoAttributes.DEBUG, "");
+    } else {
+      this.removeAttribute(MuxVideoAttributes.DEBUG);
+    }
+  }
+
+  /**
+   * Get stream type.
+   * @return {string?}
+   */
+  get streamType() {
+    return getVideoAttribute(this, MuxVideoAttributes.STREAM_TYPE);
+  }
+
+  /**
+   * Set stream type.
+   * @param  {string?} val
+   */
+  set streamType(val) {
+    this.setAttribute(MuxVideoAttributes.STREAM_TYPE, `${val}`);
+  }
+
+  /**
+   * Get the start time.
+   * @return {number}
+   */
+  get startTime() {
+    return Number(getVideoAttribute(this, MuxVideoAttributes.START_TIME));
+  }
+
+  /**
+   * Set the start time.
+   * @param  {number} val
+   */
+  set startTime(val) {
+    this.setAttribute(MuxVideoAttributes.START_TIME, `${val}`);
+  }
+
+  /**
+   * Get the preference flag for using media source.
+   * @return {boolean}
+   */
+  get preferMse() {
+    return getVideoAttribute(this, MuxVideoAttributes.PREFER_MSE) != null;
+  }
+
+  /**
+   * Set the preference flag for using media source.
+   * @param  {boolean} val
+   */
+  set preferMse(val) {
+    if (val) {
+      this.setAttribute(MuxVideoAttributes.PREFER_MSE, "");
+    } else {
+      this.removeAttribute(MuxVideoAttributes.PREFER_MSE);
+    }
+  }
+
+  /**
+   * Get the metadata object for Mux Data.
+   * @return {Metadata | undefined}
+   */
+  get metadata() {
+    return this.video?.metadata;
+  }
+
+  /**
+   * Set the metadata object for Mux Data.
+   * @param  {Metadata | undefined} val
+   */
+  set metadata(val) {
+    if (this.video) this.video.metadata = val;
+  }
+}
+
+export function getVideoAttribute(el: MuxPlayerElement, name: string) {
+  return el.video ? el.video.getAttribute(name) : el.getAttribute(name);
+}
+
+/** @TODO Refactor once using `globalThis` polyfills */
+if (!globalThis.customElements.get("mux-player")) {
+  globalThis.customElements.define("mux-player", MuxPlayerElement);
+  /** @TODO consider externalizing this (breaks standard modularity) */
+  (globalThis as any).MuxPlayerElement = MuxPlayerElement;
+}
+
```

Comment:
@cjpillsbury and I discussed this as well and we decided it's best to compose mux-video instead of inherit (the whole forest) from mux-video. It also makes more sense if we see mux-video as one of many media providers, better if this is loosely coupled in relation to the player.

Reply:
agreed 👍🏼